### ISSUE TITLE
New IdentityHashMap tests only fail with no optimization / interflow.

### ISSUE THREAD

**Issue by ekrich** [scala-native/scala-native#2429]
Symptoms can be seen in this PR. https://github.com/scala-native/scala-native/pull/2425

Note: there are some `@inline` methods in `java.util.HashMap`.
Note 2: `java.util.HashMap` is a direct port from Scala.js with only minimal changes.

**Comment by ekrich** [comment]
Failed tests:
```
org.scalanative.testsuite.javalib.util.IdentityHashMapTest.testKeySetIsViewForQueriesWithDoublesCornerCaseOfEquals
org.scalanative.testsuite.javalib.util.IdentityHashMapTest.testValuesIsViewForQueriesWithDoublesCornerCaseOfEquals
org.scalanative.testsuite.javalib.util.IdentityHashMapTest.testRemoveWithDoublesCornerCasesOfEquals
org.scalanative.testsuite.javalib.util.IdentityHashMapTest.testRemoveWithInts
org.scalanative.testsuite.javalib.util.IdentityHashMapTest.testSizeGetPutWithDoublesCornerCasesOfEquals
org.scalanative.testsuite.javalib.util.IdentityHashMapTest.testSizeGetPutWithInts
```

**Comment by WojciechMazur** [comment]
As we discussed in the mentioned PR and in the conversation on the discord 
```
About the failing tests in IdentityMap  tests - actually the behaviour we were observing was in fact correct, the tests ported from
Scala.js just used assumptions that are not correct in Scala Native, as our behaviour without optimization is matching
behaviour of the JVM. After fixing tests to run on the JVM (without ignoring IdenetityBased tests) the tests in Scala Native runs
successfully as well. However after enabling it again it would fail with 4 tests again. So here's the question should we match
JVM more closely or to provide correct answers as expected from the previous tests?
 ...
I don't think it possible to change not optimized version to worked like Scala.js does, and it is not desirable to change
optimizaiton to make it work as JVM does. In most cases expected result (depending on test) is 0/null on the JVM (or not
optimized) and some value on Scala Native (if optimized). How about we would just change the tests to check if the output is
one of the two values for Scala Native?
```
As the result logic of tests was updated according to the proposed solution and merged in #2425
